assets/js/src/tracker/index.js

```diff
@@ -0,0 +1,104 @@
+import { config } from '../config';
+
+/**
+ * A tracking utility for initializing a GA4, managing accepted
+ * events, attaching data to those events, and tracking them.
+ *
+ * @param {Object} config Configuration settings for the tracker.
+ *
+ * @return {Object} An object containing methods: init, setupEvents, event{ attach, get, track }.
+ */
+export const tracker = ( () => {
+	const events = [];
+
+	return {
+		/**
+		 * Initializes the tracker and dataLayer if not already done.
+		 */
+		init: () => {
+			if ( window[ config.tracker_var ] ) {
+				// Tracker already initialized. Do nothing.
+				return;
+			}
+
+			window.dataLayer = window.dataLayer || [];
+			window[ config.tracker_var ] = function () {
+				window.dataLayer.push( arguments );
+			};
+			window[ config.tracker_var ]( 'js', new Date() );
+			window[ config.tracker_var ](
+				'set',
+				`developer_id.${ config.developer_id }`,
+				true
+			);
+			window[ config.tracker_var ]( 'config', config.gtag_id, {
+				allow_google_signals: config.allow_google_signals,
+				link_attribution: config.link_attribution,
+				anonymize_ip: config.anonymize_ip,
+				logged_in: config.logged_in,
+				linker: config.linker,
+				custom_map: config.custom_map,
+			} );
+		},
+		/**
+		 * Each event in the array is added to a global `events` object using its `name` as the key.
+		 *
+		 * @param {Object[]} eventsArray An array of event objects.
+		 */
+		setupEvents: ( eventsArray ) => {
+			eventsArray.forEach(
+				( event ) => ( events[ event.name ] = event )
+			);
+		},
+		/**
+		 * An object to control events which exist in the global `events` object.
+		 *
+		 * @param {string} event - The name of the event.
+		 *
+		 * @return {Object} An object containing methods to attach, get, and track the event.
+		 */
+		event: ( event ) => {
+			return {
```

Comment:
:question: 
What's the purpose of creating this object every time we call `tracker.event`?
I'm not sure I follow the logic and reasoning here, plus I'm overwhelmed a bit with the number of things called "event" ;)
(global `events` array, `tracker.event`method , its `event` param, then `eventObject`s used bellow, etc.) I think we can reduce it a bit.